### ISSUE TITLE
Have TypeSafeId impl Default

### ISSUE THREAD

**Issue by Overdash** [conradludgate/type-safe-id#3]
I think it's safe to let TypeSafeId impl Default. 

`StaticType` already requires `T` to be default-able so by extension `TypeSafeId<T>` should be default-able (unless I'm missing an obvious case as to why it shouldn't).

I propose for the default impl to just call `Self::new()`. What do y'all think? Happy to raise a PR if we come to a consensus.

### Why?
Currently you cannot `#[derive(Default)] on a struct with a `TypeSafeId` field as you'll get a [E0277 error](https://doc.rust-lang.org/error_codes/E0277.html) on the field. The easiest work around is to just manually impl `Default` on the host-struct, but this is verbose and potentially unnecessary as the default impl ends up being something like:

```
type AccountId = TypeSafeId<Account>;

pub struct Account {
    pub account_id: AccountId,
}

impl Default for Account  {
    fn default() -> Self {
        Self { account_id: AccountId::new() }
    }
}
``` 

We should just be able to do 
```
type AccountId = TypeSafeId<Account>;

#[derive(Default)]
pub struct Account {
    pub account_id: AccountId,
}
```

**Comment by conradludgate** [comment]
I'm not sure how I feel about generating random values in a Default. It doesn't feel very "default" to me.

For instance, `uuid::Uuid::default()` returns `Uuid::nil()`.

**Comment by Overdash** [comment]
You're right - that's a very good point! The default implementation should be _somewhat_ deterministic. I say "somewhat" above due to the time-element of TypeIds; I don't think we can guarantee generating a totally deterministic value like `Uuid::nil()`... so I'm now a bit torn.

Since these IDs are supersets of `Uuid`, we could _also_ leverage `Uuid::nil()` as a default (i.e. in the example above `Account::from_uuid(Uuid::nil())`). However, I don't think this will return the same base32 suffix every time (I haven't tested it myself yet).
